packages/navigator/src/browser/navigator-contribution.ts

```diff
@@ -429,6 +440,51 @@ export class FileNavigatorContribution extends AbstractViewContribution<FileNavi
         this.tabbarToolbarRegistry.registerItem(item);
     };
 
+    private findTitle(tabBar: TabBar<Widget> | undefined, event?: Event): Title<Widget> | undefined {
+        if (event && event.target) {
+            let tabNode: HTMLElement | null = event.target as HTMLElement;
+            while (tabNode && !tabNode.classList.contains('p-TabBar-tab')) {
+                tabNode = tabNode.parentElement;
+            }
+            if (tabBar && tabNode && tabNode.title) {
+                let title = tabBar.titles.find(t => t.caption === tabNode!.title);
+                if (title) {
+                    return title;
+                }
+                title = tabBar.titles.find(t => t.label === tabNode!.title);
+                if (title) {
+                    return title;
+                }
+            }
+        }
+        return undefined;
+    }
+    private findTabBar(event?: Event): TabBar<Widget> | undefined {
+        if (event && event.target) {
+            const tabBar = this.shell.findWidgetForElement(event.target as HTMLElement);
+            if (tabBar instanceof TabBar) {
+                return tabBar;
+            }
+        }
+        return this.shell.currentTabBar;
+    }
+
+    /**
+     * Evaluates the widget that is triggered by the right click.
+     * @param event: `event` to be used when searching for the title and the tab-bar.
+     *
+     * @returns `widget` of the respective `title` if it exists, else returns undefined.
+     */
+    private getRequiredWidget(event?: Event): Widget | undefined {
```

Comment:
I think it would make more sense to rename the method as `getTargetWidget`.

Reply:
done.